### PR TITLE
Deprecate Environment::mergeGlobals()

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

**Review by stof — APPROVED** [review]

File: doc/deprecated.rst [hunk 1]
```diff
@@ -223,3 +223,17 @@ Testing Utilities
...
+* The ``Twig\Environment::mergeGlobals()`` method is deprecated as of Twig 3.13
+  and will be removed in Twig 4.0:
+
+  Before::
+
+      $context = $twig->mergeGlobals($context);
+
+  After::
+
+      $context += $twig->getGlobals();
```
Comment (derrabus) [inline doc/deprecated.rst:239]: `getGlobals()` is flagged as `@internal`. Shall we remove that annotation?

**Comment by ruudk** [comment]
@fabpot It is a bit unfortunate, that this was merged while Symfony TwigBridge has not yet been tagged.

Given this is included in the security release 3.14, we cannot upgrade because of deprecated usage in TwigBridge:
> Remaining indirect deprecation notices (109)
>  109x: Since twig/twig 3.14: The "Twig\Environment::mergeGlobals" method is deprecated.

Would it be an idea to get this tagged? https://github.com/symfony/symfony/pull/58207

I know one solution could be to turn off detecting of indirect deprecations, but it's super valuable to detect deprecations in vendor and help to get those fixed.

**Comment by derrabus** [comment]
@ruudk There's no shame in temporarily installing a dev snapshot of the Twig Bridge if that solves a problem for you.

**Comment by ruudk** [comment]
True. But I'm raising it, as there might be more people getting this "problem".

File: doc/deprecated.rst (hunk 1 above)
Comment (davidkmenta) [reply doc/deprecated.rst:239]: To anyone curious: it seems that `getGlobals` is `private` in v4, so the `@internal` flag makes sense.

Comment (derrabus) [reply doc/deprecated.rst:239]: But then it doesn't make sense to document that method as a replacement for a deprecated public method.

@fabpot Can you please clarify how the deprecation should be resolved downstream?

Comment (xabbuh) [reply doc/deprecated.rst:239]: Making `getGlobals()` `private` has been reverted.

Comment (davidkmenta) [reply doc/deprecated.rst:239]: @xabbuh I’m sorry, you’re right! I didn’t notice the direct commit in v4. But then @derrabus question is still valid: shouldn’t be the flag removed?

Comment (stof) [reply doc/deprecated.rst:239]: The `@internal` should indeed be removed in the 3.x branch to account for the revert of the private visibility in 4.x

Comment (fabpot) [reply doc/deprecated.rst:239]: Done in 194cced47b

Comment (derrabus) [reply doc/deprecated.rst:239]: Thanks!

**Pull request merged**